plugins/woocommerce/includes/data-stores/class-wc-product-variable-data-store-cpt.php

```diff
@@ -734,6 +734,11 @@ protected function validate_children_data( $children, $current_version ) {
 			return false;
 		}
 
+		// If we have a version set but no children at all, data is likely corrupt.
+		if ( isset( $children['version'] ) && empty( $children['all'] ) ) {
+			return false;
+		}
+
 		// Basic structure checks.
 		if ( ! isset( $children['all'] ) || ! isset( $children['visible'] ) ) {
```

Comment:
Out of curiosity, would updating the second `if` to something like this be enough?

```suggestion
		// Basic structure checks.
		if ( empty( $children['all'] ) || ! isset( $children['visible'] ) ) {
```

Or is there any reason to check `empty( $children['all'] )` specifically when `version` is set?

Reply:
I think this is an improvement I'll accept so thanks for spotting it. I _think_ I put the `version` check in there as well since I was the only who introduced this in my previous PR, so it would be validating only new transient data. However, the rest of the data structure remains the same so there's no real need to check for that anyway.